dd-sdk-android-ndk/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -34,11 +49,23 @@ std::string get_serialized_log(int signal,
                                const char *backtrace) {
     std::string serializedLog;
     serializedLog.append("{ ");
-    serializedLog.append(R"("message": ")").append(GENERIC_MESSAGE).append("\",");
-    serializedLog.append(R"("service": ")").append(SERVICE_NAME).append("\",");
-    serializedLog.append(R"("logger.name": ")").append(LOGGER_NAME).append("\",");
-    serializedLog.append(R"("env": ")").append(ENVIRONMENT).append("\",");
-    serializedLog.append(R"("status": ")").append(EMERGENCY_STATUS).append("\",");
+    if (rumcontext.application_id[0] != '\0') {
+        serializedLog.append(R"("application_id": ")").append(rumcontext.application_id)
+                .append("\",");
+    }
+    if (rumcontext.session_id[0] != '\0') {
+        serializedLog.append(R"("session_id": ")").append(rumcontext.session_id)
+                .append("\",");
+    }
+    if (rumcontext.view_id[0] != '\0') {
+        serializedLog.append(R"("view.id": ")").append(rumcontext.view_id)
+                .append("\",");
+    }
+    serializedLog.append(R"("message": ")").append(context.generic_message).append("\",");
```

Comment:
Why don't you need to check that the other fields are set like you did for the ones above?

Reply:
Because the other ones I know for sure they will be NonNull, they are marked like that in Kotlin/Jvm which starts the NDK so if something NULL will be passed will crash at runtime JVM level.

Reply:
That's a pretty good reason :-) but unclear for someone not familiar with the other part of the code. Do you think it's worth adding a comment here to explain it?